django/utils/log.py

```diff
@@ -142,6 +142,41 @@ def format_subject(self, subject):
         return subject.replace("\n", "\\n").replace("\r", "\\r")
 
 
+class DjangoSocketHandler(logging.handlers.SocketHandler):
```

Comment:
Rather, I was thinking about stripping non-picklable from `HttpRequest` directly in the `HttpRequest.__getstate__()`.

Reply:
My bad I thought you could have meant that but didn't follow through 🙇‍♂️

Reply:
So I should just add `HttpRequest.__getstate__ ()`  with the non-pickalable frozenset and depricate the `DjangoSocketHandler` class, is that right?

Reply:
There is no need to deprecate a class that never existed in Django. Just remove it from the patch.